### PR TITLE
Fix documentation warnings

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

This fixes all the warnings that were produced when generating the documentation:
```
/Users/stpasha/github/st-pasha/flame/doc/collision_detection.md:72: WARNING: 'myst' reference target not found: #Shapes
/Users/stpasha/github/st-pasha/flame/doc/collision_detection.md:113: WARNING: 'myst' reference target not found: #ScreenCollidable
/Users/stpasha/github/st-pasha/flame/doc/collision_detection.md:169: WARNING: 'myst' reference target not found: #Polygon
/Users/stpasha/github/st-pasha/flame/doc/collision_detection.md:169: WARNING: 'myst' reference target not found: #Rectangle
/Users/stpasha/github/st-pasha/flame/doc/collision_detection.md:169: WARNING: 'myst' reference target not found: #Circle
/Users/stpasha/github/st-pasha/flame/doc/collision_detection.md:185: WARNING: 'myst' reference target not found: #Collidable
/Users/stpasha/github/st-pasha/flame/doc/components.md:10: WARNING: 'myst' reference target not found: ./components.md#PositionComponent
/Users/stpasha/github/st-pasha/flame/doc/keyboard-input.md:15: WARNING: 'myst' reference target not found: #controlling-focus
/Users/stpasha/github/st-pasha/flame/doc/other-inputs.md:97: WARNING: 'myst' reference target not found: #HudButtonComponent
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:34: WARNING: 'myst' reference target not found: #BaseSystem
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:44: WARNING: 'myst' reference target not found: #RenderSystem
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:44: WARNING: 'myst' reference target not found: #UpdateSystem
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:50: WARNING: 'myst' reference target not found: #BaseSystem
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:57: WARNING: 'myst' reference target not found: #OxygenGame
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:158: WARNING: 'myst' reference target not found: #ParticleComponent
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:171: WARNING: 'myst' reference target not found: #ParticleComponent
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:184: WARNING: 'myst' reference target not found: #OxygenGame
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:234: WARNING: 'myst' reference target not found: #BaseSystem
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:259: WARNING: 'myst' reference target not found: #BaseSystem
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:284: WARNING: 'myst' reference target not found: #BaseSystem
/Users/stpasha/github/st-pasha/flame/doc/oxygen.md:367: WARNING: 'myst' reference target not found: #ParticleSystem
/Users/stpasha/github/st-pasha/flame/doc/widgets.md:23: WARNING: 'myst' reference target not found: components.md#nine-tile-box-component
```

Also some typos/grammar fixes.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: doc/collision_detection.md [hunk 1]
```diff
@@ -166,8 +166,7 @@ A Shape is the base class for representing a scalable geometrical shape. The sha
 ways of defining how they look, but they all have a size and angle that can be modified and the
 shape definition will scale or rotate the shape accordingly.
 
-There are currently three shapes: [Polygon](#Polygon), [Rectangle](#Rectangle) and
-[Cirlce](#Circle).
+There are currently three shapes: [](#polygon), [](#rectangle) and [](#circle).
```
Comment (spydon) [inline doc/collision_detection.md:169]: Shouldn't there still be text in []?

**Review by spydon — APPROVED** [review]
Nice! Just one tiny comment.

**Review by erickzanardo — APPROVED** [review]

File: doc/collision_detection.md (hunk 1 above)
Comment (st-pasha) [reply doc/collision_detection.md:169]: Turns out it's optional. I was just testing here that that is indeed so.

Comment (spydon) [reply doc/collision_detection.md:169]: But then it has the # before right?

Comment (st-pasha) [reply doc/collision_detection.md:169]: No, the `#` is only for linking. The actual title will come from the text of the header itself, which in this case is `Polygon`.

**Review by luanpotter — APPROVED** [review]
LGTM, nice!

File: doc/collision_detection.md (hunk 1 above)
Comment (spydon) [reply doc/collision_detection.md:169]: Oh, that's great!

**Pull request merged**